test/eigensolver/lobpcg_kernels.cpp

```diff
@@ -0,0 +1,196 @@
+    using Ary = gko::array<value_type>;
+    Lobpcg()
+    {
+        small_a_r = gko::initialize<Mtx_r>({{13.2, -4.3, -1.8, 0.12},
+                                            {-4.3, 24.2, -1.7, -2.3},
+                                            {-1.8, -1.7, 18.7, -0.8},
+                                            {0.12, -2.3, -0.8, 10.0}},
+                                           ref);
+        small_b_r = gko::initialize<Mtx_r>({{2.0, -1.1, 0.3, 0.01},
+                                            {-1.1, 2.5, -0.8, 0.5},
+                                            {0.3, -0.8, 2.3, -0.2},
+                                            {0.01, 0.5, -0.2, 1.9}},
+                                           ref);
+        small_a_cmplx = gko::initialize<Mtx_c>(
+            {{cmplx_value_type{13.2, 0.0}, cmplx_value_type{-4.3, -0.3},
+              cmplx_value_type{-1.8, 1.12}, cmplx_value_type{0.12, 0.6}},
+             {cmplx_value_type{-4.3, 0.3}, cmplx_value_type{24.2, 0.0},
+              cmplx_value_type{-1.7, -2.2}, cmplx_value_type{-2.3, -0.55}},
+             {cmplx_value_type{-1.8, -1.12}, cmplx_value_type{-1.7, 2.2},
+              cmplx_value_type{18.7, 0.0}, cmplx_value_type{-0.8, -1.18}},
+             {cmplx_value_type{0.12, -0.6}, cmplx_value_type{-2.3, 0.55},
+              cmplx_value_type{-0.8, 1.18}, cmplx_value_type{10.0, 0.0}}},
+            ref);
+        small_b_cmplx = gko::initialize<Mtx_c>(
+            {{cmplx_value_type{2.0, 0.0}, cmplx_value_type{-1.1, -0.1},
+              cmplx_value_type{0.3, 0.12}, cmplx_value_type{0.01, 0.4}},
+             {cmplx_value_type{-1.1, 0.1}, cmplx_value_type{2.5, 0.0},
+              cmplx_value_type{-0.8, -0.18}, cmplx_value_type{0.5, -0.097}},
+             {cmplx_value_type{0.3, -0.12}, cmplx_value_type{-0.8, 0.18},
+              cmplx_value_type{2.3, 0.0}, cmplx_value_type{-0.2, -0.172}},
+             {cmplx_value_type{0.01, -0.4}, cmplx_value_type{0.5, 0.097},
+              cmplx_value_type{-0.2, 0.172}, cmplx_value_type{1.9, 0.0}}},
+            ref);
+        small_e_vals = Ary_r(ref, 4);
+
+        d_small_a_r = gko::clone(exec, small_a_r);
+        d_small_b_r = gko::clone(exec, small_b_r);
+        d_small_a_cmplx = gko::clone(exec, small_a_cmplx);
+        d_small_b_cmplx = gko::clone(exec, small_b_cmplx);
+        d_small_e_vals = Ary_r(exec, 4);
+    }
+
+    std::shared_ptr<Mtx_r> small_a_r;
+    std::shared_ptr<Mtx_r> small_b_r;
+    std::shared_ptr<Mtx_c> small_a_cmplx;
+    std::shared_ptr<Mtx_c> small_b_cmplx;
+    std::shared_ptr<Mtx> small_a;
+    std::shared_ptr<Mtx> small_b;
+    Ary_r small_e_vals;
+
+    std::shared_ptr<Mtx_r> d_small_a_r;
+    std::shared_ptr<Mtx_r> d_small_b_r;
+    std::shared_ptr<Mtx_c> d_small_a_cmplx;
+    std::shared_ptr<Mtx_c> d_small_b_cmplx;
+    std::shared_ptr<Mtx> d_small_a;
+    std::shared_ptr<Mtx> d_small_b;
+    Ary_r d_small_e_vals;
+};
+
+TYPED_TEST_SUITE(Lobpcg, gko::test::ValueTypes, TypenameNameGenerator);
```

Comment:
```suggestion
TYPED_TEST_SUITE(Lobpcg, gko::test::ValueTypesBase, TypenameNameGenerator);
```
we have the base such that you do not support on that.
same for implementation

Reply:
Ah, thanks. That makes it a lot simpler! I changed both tests to use `ValueTypesBase`.